projects/graphql/eslint-plugin-graphql/lib/graphql-rule-adapter.js

```diff
@@ -93,54 +127,22 @@ module.exports = function adapt(rule) {
           };
 
           // this handles the scenario with visitor.enter
-          if (ruleName === 'enter') {
+          if (visitorNodeName === 'enter') {
             callbacks.enter = callback;
-          } else if (ruleName === 'leave') {
+          } else if (visitorNodeName === 'leave') {
             callbacks.leave = callback;
           } else {
-            wrappedRules[ruleName] = callback;
-          }
-        } else if (type === 'object' && rule !== null) {
-          let hadVisitor = false;
-          // bridge graphql document.leave with eslint's document:leave selector
-          if (typeof rule.enter === 'function') {
-            hadVisitor = true;
-            // eslint visitor
-            wrappedRules[ruleName] = function (node) {
-              // graphql visitor
-              return rule.enter(
-                /* node       */ context.parserServices.correspondingNode(node),
-                /* TODO: key (Not currently used by validations)  */ undefined,
-                /* parent     */ context.parserServices.correspondingNode(node.parent),
-                /* TODO: path (Not currently used by validations) */ undefined,
-                /* ancestors  */ ancestors(node).map(ancestor =>
-                  context.parserServices.correspondingNode(ancestor),
-                ),
-              );
-            };
-          }
-
-          if (typeof rule.leave === 'function') {
-            hadVisitor = true;
-            // eslint visitor
-            wrappedRules[ruleName + ':exit'] = function (node) {
-              // graphql visitor
-              return rule.leave(
-                /* node       */ context.parserServices.correspondingNode(node),
-                /* TODO: key (Not currently used by validations) */ undefined,
-                /* parent     */ context.parserServices.correspondingNode(node.parent),
-                /* TODO: path (Not currently used by validations) */ undefined,
-                /* ancestors  */ ancestors(node).map(ancestor =>
-                  context.parserServices.correspondingNode(ancestor),
-                ),
-              );
-            };
-          }
-
-          if (!hadVisitor) {
-            throw new Error(`Unsupported rule member: ${Object.keys(rule)}`);
+            // TODO: what is this case?
+            wrappedRules[visitorNodeName] = callback;
```

Comment:
hmm we should probably throw here actually; if we get an Enter/Leave visitor (i.e. a function) but it isn't enter or leave we have no idea what to do

Reply:
ok it's the enter-only case